builtin/logical/pki/storage.go

```diff
@@ -518,6 +525,56 @@ func fetchCertBundleByIssuerId(ctx context.Context, s logical.Storage, id issuer
 	return &bundle, nil
 }
 
+func writeCaBundle(ctx context.Context, s logical.Storage, caBundle *certutil.CertBundle, issuerName string, keyName string) (*issuer, *key, error) {
+	allKeyIds, err := listKeys(ctx, s)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	allIssuerIds, err := listIssuers(ctx, s)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	myKey, _, err := importKey(ctx, s, caBundle.PrivateKey, keyName)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	myIssuer, _, err := importIssuer(ctx, s, caBundle.Certificate, issuerName)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, cert := range caBundle.CAChain {
+		if _, _, err = importIssuer(ctx, s, cert, ""); err != nil {
+			return nil, nil, err
+		}
+	}
+
+	keyDefaultSet, err := isKeyDefaultSet(ctx, s)
+	if err != nil {
+		return nil, nil, err
+	}
+	if len(allKeyIds) == 0 || !keyDefaultSet {
```

Comment:
Should this just be handled by `import`? It seems cleaner than delegating to `writeBundle` logic, which won't be used in the `/import` paths for instance.

Reply:
Yup looking at this you are correct, it's been moved into the proper import methods.